src/components/dropdown/dropdown.js

```diff
@@ -97,9 +99,60 @@ class Dropdown extends mixin(createComponent, initComponentBySearch, trackBlur)
         toggle: isOfSelf && event.which !== 27 && event.which !== 40,
       };
       Object.keys(actions).forEach(action => {
+
         if (actions[action]) {
           this.element.classList[action]('bx--dropdown--open');
           this.element.focus();
+
+          /**
+           * system 360 motion
+           */
+
+          let 
+            listEl = this.element.querySelector('.bx--dropdown-list'),
+            // for iterator to calculate the total actual height
+            itemHeightTotal = 0
+          ;
+
+          // iterate through children and accumulate height
+          for(let childrenIterator = 0; childrenIterator < listEl.children.length; childrenIterator++){
+            itemHeightTotal += listEl.children[childrenIterator].offsetHeight;
```

Comment:
This line causes [forced synchronous layout](https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing#avoid_forced_synchronous_layouts). It’d be great if we don’t have to get the dimension via JavaScript. Otherwise the code should run asynchronously e.g. in [rAF](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) callback.

Reply:
Hi Asudo, I understand your concern regarding forced synchronous layout. The main reason for using this approach is to use duration that changes depending on the size of the element, which ensures smoother and consistent motion across experiences and products. I'll be committing slightly revised code for this shortly, but please let me know your thoughts on this.